pcsx2/Gif.cpp

```diff
@@ -124,7 +235,13 @@ static u32 WRITERING_DMA(u32 *pMem, u32 qwc) {
 		else
 			qwc = std::min(qwc, 8u);
 	}
-	uint size = gifUnit.TransferGSPacketData(GIF_TRANS_DMA, (u8*)pMem, qwc*16) / 16;
+	uint size;
+	if (CHECK_GIFFIFOHACK) {
+		size = gif_fifo.write(pMem, qwc);
+	}
+	else {
+		size = gifUnit.TransferGSPacketData(GIF_TRANS_DMA, (u8*)pMem, qwc * 16) / 16;
+	}
```

Comment:
It could, but there has to be a compromise when it is going to affect readability :)